libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -125,14 +110,11 @@ export class ActionSheetComponent implements AfterContentInit, AfterViewInit, On
     /** @hidden */
     private _subscriptions = new Subscription();
 
-    /** @hidden */
-    private _compact: boolean;
-
     constructor(
         private readonly _keyboardSupportService: KeyboardSupportService<ActionSheetItemComponent>,
         private readonly _changeDetectionRef: ChangeDetectorRef,
         private readonly _viewContainerRef: ViewContainerRef,
-        @Optional() private _contentDensityService: ContentDensityService,
+        @Inject(ContentDensityConsumer) private _contentDensityConsumer: ContentDensityConsumer,
```

Comment:
everywhere consumer is provided we also should mention the destroyed service so it would create its instance.